client/components/domains/register-domain-step/index.jsx

```diff
@@ -614,6 +614,12 @@ const RegisterDomainStep = React.createClass( {
 				message = translate( 'Subdomains starting with \'www.\' cannot be mapped to a WordPress.com blog' );
 				break;
 
+			case 'not_mappable_forbidden_domain':
+				message = translate( 'Sorry, you do not have the correct permissions to map %(domain)s.', {
```

Comment:
Could you be persuaded to improve this error message further? Even as a developer I was confused as to what it meant - I thought the user was not an admin, when in reality the problem was domain ownership. Something along the lines: "Sorry, only the owner of %(rootDomain)s can map its subdomains." would be more informative, IMHO.
Getting the right `rootDomain` could be tricky, but maybe just removing the first part of the domain would suffice? (since we have to be weary of second level domains like `co.uk`)